### PR TITLE
Fix extension registration order.

### LINKED ISSUE #50031: Error `Condition "int32_t(p_level) != level" is true.` when running command `godot4 --version`
### Godot version

4.0.dev.custom_build. bb409efa1

### System information

Ubuntu 21.04

### Issue description

`godot4 --version`
```
4.0.dev.custom_build.bb409efa1
ERROR: Condition "int32_t(p_level) != level" is true.
   at: deinitialize_extensions (core/extension/native_extension_manager.cpp:101)
```

### Steps to reproduce

Above

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by poiati**
Fixes https://github.com/godotengine/godot/issues/50031.

- Register core extensions right after core settings. This ensures it's properly initialized when we jump into any error branching.

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -554,6 +555,7 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	globals = memnew(ProjectSettings);
 
 	register_core_settings(); //here globals are present
+	register_core_extensions(); //before display
```
Comment (akien-mga) [inline main/main.cpp:558]: The "before display" comment is not really accurate anymore.

Comment (poiati) [reply main/main.cpp]: I agree. Let me know if my update makes sense.

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by akien-mga** [comment]
Reverted for now as this actually broke extensions: #58273.

**Pull request merged**